configure.d/1_module_mutex.conf

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+#
+# Copyright(c) 2012-2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+. $(dirname $3)/conf_framework
+
+
+check() {
+    cur_name=$(basename $2)
+    config_file_path=$1
+    if compile_module $cur_name "mutex_lock(&module_mutex);" "linux/module.h"
+    then
+        echo $cur_name 1 >> $config_file_path
+    else
+        echo $cur_name 2 >> $config_file_path
+    fi
+}
+
+apply() {
+    case "$1" in
+    "1")
+        add_define "CAS_MODULE_LOCK() \\
+            mutex_lock(&module_mutex);"
+        add_define "CAS_MODULE_UNLOCK() \\
+            mutex_unlock(&module_mutex);" ;;
+    "2")
+        add_define "CAS_MODULE_LOCK()"
```

Comment:
This empty define is ok because none of the kernels which have `module_mutex` hidden has `SYMBOL_LOOKUP_SUPPORTED` set. However the code would look cleaner if we also had define like `MODULE_MUTEX_SUPPORTED` and then guard dynamic symbol lookup with both `SYMBOL_LOOKUP_SUPPORTED` and `MODULE_MUTEX_SUPPORTED`. That way we can be sure that in case of unlikely even of enabling dynamic symbol lookup in one of future kernels we do not attempt to perform dynamic lookup without taking the `module_mutex`.

Reply:
I think I understand, I'll submit a new patch soon. Thanks, Gal.